components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/DefaultBFSIRequestObjectValidator.java

```diff
@@ -0,0 +1,149 @@
+                } else {
+                    return new ValidationResponse(false, violation);
+                }
+            } catch (RequestObjectException e) {
+                return new ValidationResponse(false, e.getMessage());
+            }
+        } else {
+            return superValidationResponse;
+        }
+    }
+
+    /**
+     * Extract clientId and scope from ob request object and check whether it's present.
+     *
+     * @param bfsiRequestObject
+     * @return result received from validateConsentIdWithClientId method
+     * @throws RequestObjectException if error occurred while validating
+     */
+    private boolean isClientIdAndScopePresent(BFSIRequestObject bfsiRequestObject) throws RequestObjectException {
+        JSONObject jsonObject = bfsiRequestObject.getSignedJWT().getPayload().toJSONObject();
+        final String clientId = jsonObject.getAsString(CLIENT_ID);
+        String scope = jsonObject.getAsString(SCOPE);
+        if (StringUtils.isBlank(clientId) || StringUtils.isBlank(scope)) {
+            log.error("Client id or scope cannot be empty");
+            throw new RequestObjectException("Client id or scope cannot be empty");
+        }
+        return true;
+    }
+
+    private String validateScope(BFSIRequestObject bfsiRequestObject, Map<String, Object> dataMap) {
+
+        try {
+            //remove scope claim
+            JWTClaimsSet claimsSet = bfsiRequestObject.getClaimsSet();
+            JSONObject claimsSetJsonObject = claimsSet.toJSONObject();
+            if (claimsSetJsonObject.containsKey("scope")) {
+                String scopeClaimString = claimsSetJsonObject.remove("scope").toString();
+                List allowedScopes = (List) dataMap.get("scope");
+                List<String> requestedScopes = new ArrayList<>(Arrays.asList(scopeClaimString.split(" ")));
+                StringBuilder stringBuilder = new StringBuilder();
+
+                // iterate through requested scopes and remove if not allowed
+                for (String scope : requestedScopes) {
+                    if (IdentityCommonConstants.OPENID_SCOPE.equals(scope)) {
+                        stringBuilder.append(scope).append(" ");
+                        if (log.isDebugEnabled()) {
+                            log.debug("Adding Openid scope to the request object");
+                        }
+                    }
+                    if (allowedScopes.contains(scope)) {
+                        stringBuilder.append(scope).append(" ");
+                        if (log.isDebugEnabled()) {
+                            log.debug(String.format("Removed scope %s from the request object",
+                                    scope.replaceAll("[\r\n]", "")));
+                        }
+                    }
+                }
+                String modifiedScopeString = stringBuilder.toString().trim();
+                // throw an error if no valid scopes found or only openid scope is found
```

Comment:
```suggestion
                // throw an error if no valid scopes found or only one scope is found (openid)
```

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/fc5b5f2d081c4d52de5072aa80c4cafa18e006b3